graph/graph.h

```diff
@@ -5,240 +5,272 @@
+    {
+        const unsigned int id = nodes.size();
+        nodes.push_back(new node(id));
+    }
 
-        void add_node()
-        {
-            int id = nodes.size();
-            nodes.push_back(new node(id));
-        }
+    void set_start(const unsigned int id)
+    {
+        if (id < nodes.size())
+            start = nodes[id];
+    }
 
-        void set_start(int id)
-        {
-            if (id < nodes.size()) 
-                start = nodes[id];
-        }
+    void set_end(const unsigned int id)
+    {
+        if (id < nodes.size())
+            end = nodes[id];
+    }
 
-        void set_end(int id)
+    void add_edge_single(const unsigned int start_id,
+                         const unsigned int end_id,
+                         const int value,
+                         const edgeType type = DIRECTIONAL)
+    {
+        if (start_id < nodes.size() && end_id < nodes.size())
         {
-            if (id < nodes.size())
-                end = nodes[id];
+            node* from = nodes[start_id];
+            node* to = nodes[end_id];
+            edge* new_edge = new
+                edge(from, to, type, new edgeDataSingle(value));
+            from->edges.push_back(new_edge);
         }
+    }
 
-        void add_edge_single(int start_id, int end_id, int value, edgeType type = DIRECTIONAL)
+    void add_edge_flow(const unsigned int start_id,
+                       const unsigned int end_id,
+                       const int flow,
+                       const int capacity,
+                       const edgeType type = DIRECTIONAL)
+    {
+        if (start_id < nodes.size() && end_id < nodes.size())
         {
-            if ((start_id < nodes.size()) && (end_id < nodes.size()))
-            {
-                node* from = nodes[start_id];
-                node* to = nodes[end_id];
-                edge* new_edge = new edge(from, to, type, new edgeDataSingle(value));
-                from->edges.push_back(new_edge); 
-            }
+            node* from = nodes[start_id];
+            node* to = nodes[end_id];
+            edge* new_edge = new edge(from,
+                                      to,
+                                      type,
+                                      new edgeDataFlow(capacity, flow));
+            from->edges.push_back(new_edge);
         }
+    }
 
-        void add_edge_flow(int start_id, int end_id, int flow, int capacity, edgeType type = DIRECTIONAL)
+    void clear_edges()
+    {
+        for (const auto& v : nodes)
         {
-            if ((start_id < nodes.size()) && (end_id < nodes.size()))
-            {
-                node* from = nodes[start_id];
-                node* to = nodes[end_id];
-                edge* new_edge = new edge(from, to, type, new edgeDataFlow(capacity, flow));
-                from->edges.push_back(new_edge);
-            }
+            v->edges.clear();
         }
+    }
 
-        void clear_edges()
+    void show(const bool res = false) const
```

Comment:
What does mean `bool res = false`?
Never seen usage `res` as an argument, always true `res == false`.
Why does it exist?